src/prefect/agent.py

```diff
@@ -167,9 +188,16 @@ async def get_work_queues(self) -> AsyncIterator[WorkQueue]:
                     except Exception:
                         self.logger.exception(f"Failed to create work queue {name!r}.")
                         continue
-
-            self._work_queue_cache.append(work_queue)
-            yield work_queue
+                else:
+                    self.logger.warning(f"Not creating missing work queue {name!r}.")
+                    work_queue = None
```

Comment:
I'm wondering why only a warning here instead of logging an exception like when we fail to create it elsewhere?

Reply:
The reason I decided to use warning was because this failure comes from an `if...else` check whereas the other failures are due to some `Exception` getting tossed. Logging an `exception` where there is none will show meaningless data.

That being said I'm open to logging this as an `error` level which would up the severity to exception but not try to print any traceback info.

Reply:
Interesting, yeah I was looking at `except ObjectNotFound`. I guess the logic in that exception is was not quite as readable to me at first. 

However, I'm wondering if "missing" is the right word. Is what is happening that "A work queue with that prefix wasn't found"? or maybe there's some clearer way to say it.

Reply:
OH I see what you mean, I think you're right that the exception isn't that readable. I just pushed a minor refactor to make the exception handling a bit easier to reason about and I changed the log level to `error` + log message to `f"Work queue '{name!r}' with prefix {self.work_queue_prefix} wasn't found"`